apps/emqx_dashboard/src/emqx_dashboard_monitor.erl

```diff
@@ -360,18 +356,48 @@ cal_rate_(Key, {Now, Last, TDelta, Res}) ->
     RateKey = maps:get(Key, ?DELTA_SAMPLER_RATE_MAP),
     {Now, Last, TDelta, Res#{RateKey => Rate}}.
 
-granularity_adapter([], Res) ->
-    lists:reverse(Res);
-granularity_adapter([Sampler], Res) ->
-    granularity_adapter([], [Sampler | Res]);
-granularity_adapter([Sampler1, Sampler2 | Rest], Res) ->
-    Fun =
-        fun(Key, M) ->
-            Value1 = maps:get(Key, Sampler1),
-            Value2 = maps:get(Key, Sampler2),
-            M#{Key => Value1 + Value2}
-        end,
-    granularity_adapter(Rest, [lists:foldl(Fun, Sampler2, ?DELTA_SAMPLER_LIST) | Res]).
+%% Try to keep the total number of recrods around 1000.
+%% When the oldest data point is
+%% < 1h: sample every 10s: 360 data points
```

Comment:
Q: Why not 5s?

Reply:
current sample tick is every 10s, i do not intend to make it tick more often.